### DISCUSSION TITLE
`pylsp` configuration

### DISCUSSION THREAD

**Discussion by nabiu256** [VonHeikemen/lsp-zero.nvim#80] [Q&A] [answered]
I've tried configuring the `python-lsp-server` so that it uses flake8 instead of pycodestyle and the other defaults by following the instructions on their repo ([here](https://github.com/python-lsp/python-lsp-server#configuration)). I've succedeed at disabling `pycodestyle` and the others by doing

```lua
pycodestyle = {
    enabled = false,
}
```
but trying to set `enabled = true` and `configurationSources = {'flake8'}` doesn't seem to be doing anything, I just end up with no linter whenever I open a python file (although the lsp server definitely attaches, I've checked).

Does anyone have any idea on what might I be doing wrong?

**Comment by VonHeikemen** [comment, marked as answer]
I tried to look for examples and found [this old post](https://waylonwalker.com/setup-pylsp/). Translated the options to lsp-zero.

```lua
local lsp = require('lsp-zero')
lsp.preset('recommended')

lsp.configure('pylsp', {
  settings = {
    pylsp = {
      configurationSources = {'flake8'},
      plugins = {
        pycodestyle = {enabled = false},
        flake8 = {
          enabled = true,
          ignore = {},
        }
      }
    }
  }
})

lsp.setup()
```

**Reply by nabiu256** [reply]
This is mostly what I had already tried, and while I'm still facing some problems, it does work! Thank you.

Edit: Fixed my mistake, I simply had not activated my venv and pylsp couldn't access flake8.
